gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/TaskManagerController.java

```diff
@@ -179,4 +181,20 @@ public ResponseEntity<byte[]> getLog(@PathVariable String timestamp) throws IOEx
     private String removeIllegalUrlCharacter(String s) {
         return s.replace(":", "");
     }
+
+    @PostMapping(value = "/tasks/{timestamp}/uploadfile")
+    public ResponseEntity<Object> uploadFile(@PathVariable String timestamp,
+                                             @RequestParam("file") MultipartFile file,
+                                             @RequestParam("fileName") String fileName,
```

Comment:
We can also check the filename to prevent path injection or other vulnerabilities:
if (!fileName.matches("^[a-zA-Z0-9-_]+\\.\\w+$")) {
    throw new IllegalArgumentException("Invalid file name");
}